shanoir-ng-datasets/src/main/java/org/shanoir/ng/importer/strategies/dataset/CtDatasetStrategy.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Shanoir NG - Import, manage and share neuroimaging data
+ * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
+ * Contact us on https://project.inria.fr/shanoir/
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+package org.shanoir.ng.importer.strategies.dataset;
+
+import org.dcm4che3.data.Attributes;
+import org.shanoir.ng.dataset.modality.CtDataset;
+import org.shanoir.ng.dataset.model.CardinalityOfRelatedSubjects;
+import org.shanoir.ng.dataset.model.DatasetExpression;
+import org.shanoir.ng.dataset.model.DatasetMetadata;
+import org.shanoir.ng.dataset.model.DatasetModalityType;
+import org.shanoir.ng.dataset.model.ProcessedDatasetType;
+import org.shanoir.ng.dicom.DicomProcessing;
+import org.shanoir.ng.importer.dto.Dataset;
+import org.shanoir.ng.importer.dto.DatasetsWrapper;
+import org.shanoir.ng.importer.dto.ExpressionFormat;
+import org.shanoir.ng.importer.dto.ImportJob;
+import org.shanoir.ng.importer.dto.Serie;
+import org.shanoir.ng.importer.strategies.datasetexpression.DatasetExpressionContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CtDatasetStrategy implements DatasetStrategy<CtDataset> {
+
+	@Autowired
+	DicomProcessing dicomProcessing;
+	
+	@Autowired
+	DatasetExpressionContext datasetExpressionContext;
+	
+	@Override
+	public DatasetsWrapper<CtDataset> generateDatasetsForSerie(Attributes dicomAttributes, Serie serie,
+			ImportJob importJob) throws Exception {
+		
+		DatasetsWrapper<CtDataset> datasetWrapper = new DatasetsWrapper<>();
+		/**
+		 * retrieve number of dataset in current serie if Number of dataset > 1 then
+		 * each dataset will be named with an int at the end of the name. else the is
+		 * only one dataset => no need for extension.
+		 */
+		int datasetIndex;
+		if (serie.getDatasets().size() > 1) {
+			datasetIndex = 1;
+		} else {
+			datasetIndex = -1;
+		}
+
```

Comment:
I think it's quite a lot of copy/paste, but maybe you can:
- Remove MR comments (and TODOs..)
- Remove the first initialization "CtDataset dataset = new CtDataset();" => It is replaced later

Reply:
Done